src/Utils/Helpers.ts

```diff
@@ -88,4 +89,14 @@ const isInComponents = (path: string): boolean => {
 const isInChannels = (path: string): boolean => {
   return path.startsWith('channels.');
 };
-export { compareComponents, isEqual, isInComponents, isInChannels };
+const toJS = (asyncapiYAMLorJSON: any): any => {
+  if (asyncapiYAMLorJSON.constructor && asyncapiYAMLorJSON.constructor.name === 'Object') {
+    //NOTE: this approach can have problem with circular references between object and JSON.stringify doesn't support it.
+    //more info: https://github.com/asyncapi/parser-js/issues/293
+    return JSON.parse(JSON.stringify(asyncapiYAMLorJSON));
+  } 
+  if (typeof asyncapiYAMLorJSON === 'string') {
+    return YAML.load(asyncapiYAMLorJSON);
+  }
```

Comment:
Last suggestion: Please throw error, if someone pass to the Optimizer the other value than string/JSON object.